puffin/src/agent.rs

```diff
@@ -219,8 +220,8 @@ impl<PB: ProtocolBehavior> Agent<PB> {
 }
 
 impl<PB: ProtocolBehavior> Stream<PB> for Agent<PB> {
-    fn add_to_inbound(&mut self, message_flight: &PB::OpaqueProtocolMessageFlight) {
-        self.put.add_to_inbound(message_flight)
+    fn add_to_inbound(&mut self, message: &ConcreteMessage) {
```

Comment:
Consider making `take_message_from_outbound` outputs a ConcreteMessage too so that:
 -  we can get rid of protocol-dependent boiler-plate code for reading the buffer into a `OpaqueProtocolMessageFlight`
  - and Stream does not depend on `ProtocolBehavior` anymore